source/common/config/subscription_factory_impl.cc

```diff
@@ -56,28 +57,51 @@ SubscriptionPtr SubscriptionFactoryImpl::subscriptionFromConfigSource(
           restMethod(type_url, transport_api_version), type_url, callbacks, resource_decoder, stats,
           Utility::configSourceInitialFetchTimeout(config), validation_visitor_);
-    case envoy::config::core::v3::ApiConfigSource::GRPC:
+    case envoy::config::core::v3::ApiConfigSource::GRPC: {
+      GrpcMuxSharedPtr mux;
+      if (Runtime::runtimeFeatureEnabled("envoy.reloadable_features.unified_mux")) {
+        mux = std::make_shared<Config::XdsMux::GrpcMuxSotw>(
+            Utility::factoryForGrpcApiConfigSource(cm_.grpcAsyncClientManager(), api_config_source,
+                                                   scope, true)
+                ->createUncachedRawAsyncClient(),
+            dispatcher_, sotwGrpcMethod(type_url, transport_api_version), api_.randomGenerator(),
+            scope, Utility::parseRateLimitSettings(api_config_source), local_info_,
+            api_config_source.set_node_on_first_message_only());
+      } else {
+        mux = std::make_shared<Config::GrpcMuxImpl>(
+            local_info_,
+            Utility::factoryForGrpcApiConfigSource(cm_.grpcAsyncClientManager(), api_config_source,
+                                                   scope, true)
+                ->createUncachedRawAsyncClient(),
+            dispatcher_, sotwGrpcMethod(type_url, transport_api_version), api_.randomGenerator(),
+            scope, Utility::parseRateLimitSettings(api_config_source),
+            api_config_source.set_node_on_first_message_only());
+      }
       return std::make_unique<GrpcSubscriptionImpl>(
-          std::make_shared<Config::GrpcMuxImpl>(
-              local_info_,
-              Utility::factoryForGrpcApiConfigSource(cm_.grpcAsyncClientManager(),
-                                                     api_config_source, scope, true)
-                  ->createUncachedRawAsyncClient(),
-              dispatcher_, sotwGrpcMethod(type_url, transport_api_version), api_.randomGenerator(),
-              scope, Utility::parseRateLimitSettings(api_config_source),
-              api_config_source.set_node_on_first_message_only()),
-          callbacks, resource_decoder, stats, type_url, dispatcher_,
+          std::move(mux), callbacks, resource_decoder, stats, type_url, dispatcher_,
           Utility::configSourceInitialFetchTimeout(config),
           /*is_aggregated*/ false, options);
+    }
     case envoy::config::core::v3::ApiConfigSource::DELTA_GRPC: {
+      GrpcMuxSharedPtr mux;
+      if (Runtime::runtimeFeatureEnabled("envoy.reloadable_features.unified_mux")) {
+        mux = std::make_shared<Config::XdsMux::GrpcMuxDelta>(
+            Utility::factoryForGrpcApiConfigSource(cm_.grpcAsyncClientManager(), api_config_source,
+                                                   scope, true)
+                ->createUncachedRawAsyncClient(),
+            dispatcher_, sotwGrpcMethod(type_url, transport_api_version), api_.randomGenerator(),
```

Comment:
Why is this SotW for the delta case?

Reply:
No good reason, fixed.

Reply:
How come this wasn't caught be a test?

Reply:
So the issue with `test/common/config/subscription_factory_impl_test.cc` is that none of the tests can actually catch this as written: the tests either do not start subscription that was created as part of the test, or use a NiceMock for the sync_client and do not verify parameters of the `sync_client.start()` call.

I also forgot to update vhds_integration_test to also test against the unified mux (fixed), which would've failed.